mycroft/client/speech/mic.py

```diff
@@ -303,11 +305,22 @@ def _wait_until_wake_word(self, source, sec_per_buffer):
         counter = 0
 
         while not said_wake_word:
-            if check_for_signal('buttonPress'):
-                logger.debug("Button Pressed, wakeword not needed")
+            # Check if told programatically to skip the wake word for dialog
+            if check_for_signal('startListening'):
                 said_wake_word = True
                 continue
 
+            # Pressing the button can start recording (unless it is being
+            # used to mean 'stop' instead)
+            if check_for_signal('buttonPress', 1):
+                # give others time to consume this as 'stop'
+                sleep(0.25)
+                if check_for_signal('buttonPress'):
+                    # signal is still here, assume it was to begin recording
+                    logger.debug("Button Pressed, wakeword not needed")
+                    said_wake_word = True
+                    continue
+
```

Comment:
Is this done in preparation for future architecture change? To me it looks like the buttonPress can't be used in two places at once. The only other place is in series with this check. (while recording to send to STT)

In any case we might want to move this to a separate function `check_for_button()` (or similar) and have the if statement at L309 check both criterias
`if check_for_signal('startListening') or check_for_button():`

The `_wait_for_wakeword()` method is complex enough as it is without logic for whether the buttonPressed signal means stop or start recording.